server/src/utils.ts

```diff
@@ -72,17 +72,30 @@ export let findBscNativeOfFile = (
 
 // TODO: this doesn't handle file:/// scheme
 export let findNodeBuildOfProjectRoot = (
-  projectRootPath: p.DocumentUri
+  projectRootPath: p.DocumentUri,
+  pathToBinFromConfig: p.DocumentUri
 ): null | { buildPath: p.DocumentUri; isReScript: boolean } => {
-  let rescriptNodePath = path.join(projectRootPath, c.rescriptNodePartialPath);
-  let bsbNodePath = path.join(projectRootPath, c.bsbNodePartialPath);
+  if (pathToBinFromConfig !== "") {
+    if (fs.existsSync(pathToBinFromConfig)) {
+      if (pathToBinFromConfig.endsWith(c.rescriptBinName)) {
```

Comment:
I'm confused here: it seems the config either sets the path to `bsb` or to `rescript`. Should it not give the path to the directory that contains those binaries instead?

Also, what about formatting? That uses the `bsc` command which does not seem to be considered here.

Reply:
I looked at this issue #412 , and there was an example of using the path to the binary. But yes, maybe we should use the directory path.

Reply:
fixed